src/thor/route_action.cc

```diff
@@ -314,54 +314,55 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
 
     // Get best path and keep it
     auto temp_paths = get_path(path_algorithm, *origin, *destination, costing, api.options());
-    auto& temp_path = temp_paths.front();
-    first_edge = temp_path.front().edgeid;
-    temp_path.swap(path); // so we can append to path instead of prepend
-
-    // Merge through legs by updating the time and splicing the lists
-    if (!temp_path.empty()) {
-      auto offset = path.back().elapsed_time;
-      std::for_each(temp_path.begin(), temp_path.end(),
-                    [offset](PathInfo& i) { i.elapsed_time += offset; });
-      // Connects via the same edge so we only need it once
-      if (path.back().edgeid == temp_path.front().edgeid) {
-        path.pop_back();
-      } else if (destination->type() == valhalla::Location::kVia) {
-        // Insert a route discontinuity if the paths meet at opposing edges and not
-        // at a graph node. Use path size - 1 as the index where the discontinuity lies.
-        via_discontinuity(*reader, *destination, path.back().edgeid, temp_path.front().edgeid, vias,
-                          temp_path.size(), true);
-      }
-      path.insert(path.end(), temp_path.begin(), temp_path.end());
-    }
+    for (auto& temp_path : temp_paths) {
```

Comment:
this is a 2 line change, the rest is just whitespace. the first change is here to add the loop over the paths instead of just taking the first one